OSSM-5832 Changes to being able to run test on arm cluster

fjglira opened the pull request:
Jira: https://issues.redhat.com/browse/OSSM-5832

Whit this small changes we will be able to run the test against arm cluster by duplicating the scenarios that we currently have but with the arm configuration

fjglira committed:
Initial changes to being able to run test on arm cluster

Add minor changes

Add explicit compare to arch arm64

jwendell commented on tests/integration/common-operator-integ-suite.sh:
```diff
@@ -40,6 +41,10 @@ parse_flags() {
         shift
         SKIP_BUILD=true
         ;;
+      --arm64)
```
why do we need to pass a parameter? Can't the script figure out the platform itself?

fjglira replied:
I do it but in the specific ocp script here: https://github.com/maistra/istio-operator/blob/365adca0c556d59ec3f330b5b1464a144e717da9/tests/integration/integ-suite-ocp.sh#L23-L28
To have it outside the common file for those types of validation, but I can add inside of common file if make more sense

fjglira replied:
Done

fjglira committed:
Update the validation to check arch of the cluster

Fix un set variable

Delete unused variable

bmangoen commented on Makefile.core.mk:
```diff
@@ -168,7 +171,7 @@ run: gen ## Run a controller from your host.
 # docker build -t ${IMAGE} --build-arg GIT_TAG=${GIT_TAG} --build-arg GIT_REVISION=${GIT_REVISION} --build-arg GIT_STATUS=${GIT_STATUS} .
 .PHONY: docker-build
 docker-build: build ## Build docker image with the manager.
```
Not sure it would work because the `build` target uses `build-amd64` (cf. [here](https://github.com/maistra/istio-operator/blob/maistra-3.0/Makefile.core.mk#L162)). That means the arm64 container image would use the wrong arch of the manager binary.

fjglira replied:
I made the change only for the target docker-build and I tested on my fedora VM with amd64 and works. The operator was deployed successfully on arm OCP cluster

fjglira replied:
This is because I set --platform=linux/arm64 flag and value for the docker build command

dgn commented on Makefile.core.mk:
```diff
@@ -258,7 +261,8 @@ deploy-yaml: helm ## Outputs YAML manifests needed to deploy the controller
 deploy-openshift: helm ## Deploy controller to OpenShift via YAML manifests
 	$(info NAMESPACE: $(NAMESPACE))
-	$(MAKE) -s deploy-yaml-openshift | kubectl apply -f -
+	export HELM_TEMPL_DEF_FLAGS="--include-crds --set image='${IMAGE}'" && \
```
can we split this out into a separate commit and fix it everywhere?

fjglira replied:
Sure, I'll do that

fjglira committed:
Merge branch 'maistra-3.0' into OSSM-5832

fjglira committed:
Update make target to use build or build-arm

Update changes to work on different arch

Update changes to work on different arch

fjglira committed:
Adding decalration to DOCKER_BUILD_FLAGS

fjglira commented:
/retest

fjglira commented:
/retest

dgn commented on Makefile.core.mk:
```diff
@@ -68,6 +68,9 @@ NAMESPACE ?= istio-operator
 ENVTEST_K8S_VERSION = 1.26.0
 
+# Set DOCKER_BUILD_FLAGS to specify flags to pass to 'docker build', default to empty. Example: --platform=linux/arm64
+DOCKER_BUILD_FLAGS ?= "--platform=$(TARGET_OS)/$(TARGET_ARCH)"
```
we don't seem to set TARGET_OS and TARGET_ARCH by default- should we?

fjglira replied:
is set by this script by default `/common/scripts/setup_env.sh`

dgn replied:
oh I see! thanks, nevermind then
